### PR TITLE
[ADD] udes_scanned_by_tracking: Initial commit

### PR DESCRIPTION
**Pull request by nocturnalastro**
Signed-off-by: Michele Costa <michele.costa@unipart.io>

### CODE REVIEW

File: addons/udes_scanned_by_tracking/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+{
+    "name": "UDES Scanned By Tracking",
+    "version": "11.0",
+    "summary": "Inventory, Logistics, Warehousing",
+    "description": "Extension of udes_stock model for UDES, to add the user to a move line",
+    "depends": ["stock",],
+    "category": "Warehouse",
+    "version": "0.1",
```
Comment (PeterC10) [inline addons/udes_scanned_by_tracking/__manifest__.py:9]: `version` is listed twice on lines 4 and 9

File: addons/udes_scanned_by_tracking/models/stock_move_line.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        "Completion datetime", help="Date and time the operation was completed.", index=True
+    )
+
+    @staticmethod
+    def _now_for_tracking_data():
+        """Can be overwritten to change the tracking resolution"""
+        return datetime.now()
+
+    @api.model
+    def _add_user_tracking_data(self, vals):
```
Comment (PeterC10) [inline addons/udes_scanned_by_tracking/models/stock_move_line.py:28]: Nit: Do we need a docstring here?

File: addons/udes_scanned_by_tracking/__manifest__.py (hunk 1 above)
Comment (nocturnalastro) [reply addons/udes_scanned_by_tracking/__manifest__.py:9]: Copy pasta error :facepalm:

Comment (nocturnalastro) [reply addons/udes_scanned_by_tracking/__manifest__.py]: Fixed: stuck with 11.0 as to track the odoo version is sould be against

File: addons/udes_scanned_by_tracking/models/stock_move_line.py (hunk 2 above)
Comment (nocturnalastro) [reply addons/udes_scanned_by_tracking/models/stock_move_line.py:28]: Fixed

**Review by PeterC10 — APPROVED** [review]
LGTM

File: addons/udes_scanned_by_tracking/models/stock_move_line.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        """Inject/overwrite user tracking data into vals"""
+
+        if vals.get("qty_done", 0) > 0:
+            if not self.env.real_uid:
```
Comment (ws-mps) [inline addons/udes_scanned_by_tracking/models/stock_move_line.py:32]: Worth adding a comment that this field is expected to be added in Odoo core.

Comment (nocturnalastro) [reply addons/udes_scanned_by_tracking/models/stock_move_line.py]: Fixed

**Review by ws-mps — APPROVED** [review]

**Pull request merged**